### PR TITLE
Add `progress_callback` parameter to `KubernetesPodOperator`

### LINKED ISSUE #31745: Add a process_line callback to KubernetesPodOperator
### Description

Add a process_line callback to KubernetesPodOperator 
Like https://github.com/apache/airflow/blob/main/airflow/providers/apache/beam/operators/beam.py#LL304C36-L304C57 the `BeamRunPythonPipelineOperator`, which allows the user to add stateful plugins based on the logging from docker job 

### Use case/motivation

I can add a plugin based on the logging and also allow cleanup on_kill based on the job creation log. 

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by zeotuan**
closes: #31745 
add progress_callback to allow custom behavior based on job logs

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Nice and simple.

**Comment by potiuk** [comment]
I recommend installing pre-commit and rebasing/amending the change - it will then fix the static checks for you.

File: airflow/providers/cncf/kubernetes/operators/pod.py [hunk 1]
```diff
@@ -328,6 +328,7 @@ def __init__(
         is_delete_operator_pod: None | bool = None,
         termination_message_policy: str = "File",
         active_deadline_seconds: int | None = None,
+        progress_callback: Callable[[str], None] | None = None,
```
Comment (eladkal) [inline airflow/providers/cncf/kubernetes/operators/pod.py:332]: Please add this parameter to the class doc string description 
`:param progress_callback: ...`

File: airflow/providers/cncf/kubernetes/utils/pod_manager.py [hunk 2]
```diff
@@ -413,6 +416,9 @@ def consume_logs(
                 for raw_line in logs:
                     line = raw_line.decode("utf-8", errors="backslashreplace")
                     line_timestamp, message = self.parse_log_line(line)
+                    if self._progress_callback:
+                        print(line)
```
Comment (hussein-awala) [inline airflow/providers/cncf/kubernetes/utils/pod_manager.py]: why do you need to print the line? I think you should remove this line:
```suggestion
```

**Review by hussein-awala — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**